packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -386,11 +411,21 @@ class _CupertinoCheckboxState extends State<CupertinoCheckbox> with TickerProvid
           .withSaturation(kCupertinoFocusColorSaturation)
           .toColor();
 
+    final WidgetStateProperty<MouseCursor> effectiveMouseCursor =
+      WidgetStateProperty.resolveWith<MouseCursor>((Set<WidgetState> states) {
+        return WidgetStateProperty.resolveAs<MouseCursor?>(widget.mouseCursor, states)
+          ?? (states.contains(WidgetState.disabled)
+              ? SystemMouseCursors.basic
+              : kIsWeb ? SystemMouseCursors.click : SystemMouseCursors.basic
+            );
```

Comment:
Nit: Would a `switch` be more readable than this double ternary?

Reply:
I can't immediately see a way to do this. @nate-thegrate do you had any ideas?

Reply:
What about `kIsWeb && !states.contains(WidgetState.disabled) ? SystemMouseCursors.click : SystemMouseCursors.basic`?